pkg/cmd/experimental/syncgroups/ad/ldapinterface_test.go

```diff
@@ -0,0 +1,202 @@
+package ad
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"gopkg.in/ldap.v2"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	"github.com/openshift/origin/pkg/auth/ldaputil/testclient"
+)
+
+func newTestADLDAPInterface(client ldap.Client) *ADLDAPInterface {
```

Comment:
When you're making things like this in the future.  `BaseDN` is likely to be a string that test needs to know about and might care to change.  Take it as an argument.  Same one could be passed to `newUser` and the proper DN could be built.  This is good enough (with `newTestUser`) that I'm willing to merge it.